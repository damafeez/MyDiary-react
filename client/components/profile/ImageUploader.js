import React, { Component } from 'react';
import icons from '../../assets/icons.svg';
import { userPlaceholderImage } from '../../mixin';
import './ImageUploader.scss';

class ProfileImageUploader extends Component {
  state = {
    imagePreview: '',
  }

  handleChange = (e) => {
    const file = e.target.files[0];
    e.preventDefault();
    const reader = new FileReader();
    reader.onloadend = () => {
      this.upload(file);
      this.setState({ imagePreview: reader.result });
    };
    if (file) reader.readAsDataURL(file);
  }

  upload = (image) => {
    const { uploadImage } = this.props;
    const imgFile = new FormData();
    imgFile.append('profileImage', image);
    uploadImage(imgFile);
  }

  render() {
    const {
      profileImage,
      loading,
    } = this.props;
    const { imagePreview } = this.state;
    return (
      <div
        className="profile-image profile-image-uploader"
        style={{ backgroundImage: `linear-gradient(rgba(150, 150, 150, 0.4), rgba(150, 150, 150, 0.4)), url(${imagePreview || profileImage || userPlaceholderImage})` }}
      >
        {profileImage && <span onClick={() => this.upload(null)} className="profile-image-uploader_cancel hoverable centralizer">x</span>}
        <input type="file" onChange={this.handleChange} id="uploader" accept=".png, .jpg, .jpeg" />
        <label htmlFor="uploader">
          <svg className={`icon ${loading ? 'spin' : ''}`}>
            <use xlinkHref={`${icons}#camera`} />
          </svg>
        </label>
      </div>
    );
  }
}

export default ProfileImageUploader;
