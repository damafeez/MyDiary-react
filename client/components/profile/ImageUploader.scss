.profile-image-uploader {
  background: linear-gradient(rgba(gray, .4), rgba(gray, .4)), url(https://res.cloudinary.com/drmmqcxkc/image/upload/v1541505049/user.jpg);
  background-size: cover;
  position: relative;
  &_cancel {
    position: absolute;
    cursor: pointer;
    right: 10px;
    top: 0;
    width: 20px;
    height: 20px;
    border-radius: 50%;
    z-index: 5;
    box-shadow: 1px 5px 15px rgba(0, 0, 0, 0.2);
    color: red;
    background: whitesmoke;
  }
  input {
    display: none;
  }
  .icon {
    width: 24px;
    height: 24px;
    padding: 25px;
    box-sizing: content-box;
    cursor: pointer;
  }
}
