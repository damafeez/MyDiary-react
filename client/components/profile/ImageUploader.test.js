import React from 'react';
import ImageUploader from './ImageUploader';

const setup = (overrideProps) => {
  const props = {
    value: 'name',
  };
  const wrapper = shallow(<ImageUploader {...props} {...overrideProps} />);
  return { props, wrapper };
};

it('matches snapshot', () => {
  const { wrapper } = setup();
  expect(wrapper).toMatchSnapshot();
});
it('matches snapshot:::loading:', () => {
  const { wrapper } = setup({ loading: true });
  expect(wrapper).toMatchSnapshot();
});
it('matches snapshot:::profileImage', () => {
  const { wrapper } = setup({ profileImage: 'profileImage' });
  expect(wrapper).toMatchSnapshot();
});
