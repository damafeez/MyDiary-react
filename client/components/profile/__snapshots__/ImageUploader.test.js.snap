// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfileImageUploader
    value="name"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          accept=".png, .jpg, .jpeg"
          id="uploader"
          onChange={[Function]}
          type="file"
        />,
        <label
          htmlFor="uploader"
        >
          <svg
            className="icon "
          >
            <use
              xlinkHref="[object Object]#camera"
            />
          </svg>
        </label>,
      ],
      "className": "profile-image profile-image-uploader",
      "style": Object {
        "backgroundImage": "linear-gradient(rgba(150, 150, 150, 0.4), rgba(150, 150, 150, 0.4)), url(https://res.cloudinary.com/drmmqcxkc/image/upload/v1541506955/user-placeholder.png)",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "accept": ".png, .jpg, .jpeg",
          "disabled": undefined,
          "id": "uploader",
          "onChange": [Function],
          "type": "file",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <svg
            className="icon "
          >
            <use
              xlinkHref="[object Object]#camera"
            />
          </svg>,
          "htmlFor": "uploader",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <use
              xlinkHref="[object Object]#camera"
            />,
            "className": "icon ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "xlinkHref": "[object Object]#camera",
            },
            "ref": null,
            "rendered": null,
            "type": "use",
          },
          "type": "svg",
        },
        "type": "label",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            accept=".png, .jpg, .jpeg"
            id="uploader"
            onChange={[Function]}
            type="file"
          />,
          <label
            htmlFor="uploader"
          >
            <svg
              className="icon "
            >
              <use
                xlinkHref="[object Object]#camera"
              />
            </svg>
          </label>,
        ],
        "className": "profile-image profile-image-uploader",
        "style": Object {
          "backgroundImage": "linear-gradient(rgba(150, 150, 150, 0.4), rgba(150, 150, 150, 0.4)), url(https://res.cloudinary.com/drmmqcxkc/image/upload/v1541506955/user-placeholder.png)",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "accept": ".png, .jpg, .jpeg",
            "disabled": undefined,
            "id": "uploader",
            "onChange": [Function],
            "type": "file",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <svg
              className="icon "
            >
              <use
                xlinkHref="[object Object]#camera"
              />
            </svg>,
            "htmlFor": "uploader",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <use
                xlinkHref="[object Object]#camera"
              />,
              "className": "icon ",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "xlinkHref": "[object Object]#camera",
              },
              "ref": null,
              "rendered": null,
              "type": "use",
            },
            "type": "svg",
          },
          "type": "label",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`matches snapshot:::loading: 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfileImageUploader
    loading={true}
    value="name"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          accept=".png, .jpg, .jpeg"
          disabled={true}
          id="uploader"
          onChange={[Function]}
          type="file"
        />,
        <label
          htmlFor="uploader"
        >
          <svg
            className="icon spin"
          >
            <use
              xlinkHref="[object Object]#camera"
            />
          </svg>
        </label>,
      ],
      "className": "profile-image profile-image-uploader",
      "style": Object {
        "backgroundImage": "linear-gradient(rgba(150, 150, 150, 0.4), rgba(150, 150, 150, 0.4)), url(https://res.cloudinary.com/drmmqcxkc/image/upload/v1541506955/user-placeholder.png)",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "accept": ".png, .jpg, .jpeg",
          "disabled": true,
          "id": "uploader",
          "onChange": [Function],
          "type": "file",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <svg
            className="icon spin"
          >
            <use
              xlinkHref="[object Object]#camera"
            />
          </svg>,
          "htmlFor": "uploader",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <use
              xlinkHref="[object Object]#camera"
            />,
            "className": "icon spin",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "xlinkHref": "[object Object]#camera",
            },
            "ref": null,
            "rendered": null,
            "type": "use",
          },
          "type": "svg",
        },
        "type": "label",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            accept=".png, .jpg, .jpeg"
            disabled={true}
            id="uploader"
            onChange={[Function]}
            type="file"
          />,
          <label
            htmlFor="uploader"
          >
            <svg
              className="icon spin"
            >
              <use
                xlinkHref="[object Object]#camera"
              />
            </svg>
          </label>,
        ],
        "className": "profile-image profile-image-uploader",
        "style": Object {
          "backgroundImage": "linear-gradient(rgba(150, 150, 150, 0.4), rgba(150, 150, 150, 0.4)), url(https://res.cloudinary.com/drmmqcxkc/image/upload/v1541506955/user-placeholder.png)",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "accept": ".png, .jpg, .jpeg",
            "disabled": true,
            "id": "uploader",
            "onChange": [Function],
            "type": "file",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <svg
              className="icon spin"
            >
              <use
                xlinkHref="[object Object]#camera"
              />
            </svg>,
            "htmlFor": "uploader",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <use
                xlinkHref="[object Object]#camera"
              />,
              "className": "icon spin",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "xlinkHref": "[object Object]#camera",
              },
              "ref": null,
              "rendered": null,
              "type": "use",
            },
            "type": "svg",
          },
          "type": "label",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`matches snapshot:::profileImage 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfileImageUploader
    profileImage="profileImage"
    value="name"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          accept=".png, .jpg, .jpeg"
          id="uploader"
          onChange={[Function]}
          type="file"
        />,
        <label
          htmlFor="uploader"
        >
          <svg
            className="icon "
          >
            <use
              xlinkHref="[object Object]#camera"
            />
          </svg>
        </label>,
      ],
      "className": "profile-image profile-image-uploader",
      "style": Object {
        "backgroundImage": "linear-gradient(rgba(150, 150, 150, 0.4), rgba(150, 150, 150, 0.4)), url(profileImage)",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "accept": ".png, .jpg, .jpeg",
          "disabled": undefined,
          "id": "uploader",
          "onChange": [Function],
          "type": "file",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <svg
            className="icon "
          >
            <use
              xlinkHref="[object Object]#camera"
            />
          </svg>,
          "htmlFor": "uploader",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <use
              xlinkHref="[object Object]#camera"
            />,
            "className": "icon ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "xlinkHref": "[object Object]#camera",
            },
            "ref": null,
            "rendered": null,
            "type": "use",
          },
          "type": "svg",
        },
        "type": "label",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            accept=".png, .jpg, .jpeg"
            id="uploader"
            onChange={[Function]}
            type="file"
          />,
          <label
            htmlFor="uploader"
          >
            <svg
              className="icon "
            >
              <use
                xlinkHref="[object Object]#camera"
              />
            </svg>
          </label>,
        ],
        "className": "profile-image profile-image-uploader",
        "style": Object {
          "backgroundImage": "linear-gradient(rgba(150, 150, 150, 0.4), rgba(150, 150, 150, 0.4)), url(profileImage)",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "accept": ".png, .jpg, .jpeg",
            "disabled": undefined,
            "id": "uploader",
            "onChange": [Function],
            "type": "file",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <svg
              className="icon "
            >
              <use
                xlinkHref="[object Object]#camera"
              />
            </svg>,
            "htmlFor": "uploader",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <use
                xlinkHref="[object Object]#camera"
              />,
              "className": "icon ",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "xlinkHref": "[object Object]#camera",
              },
              "ref": null,
              "rendered": null,
              "type": "use",
            },
            "type": "svg",
          },
          "type": "label",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
